### PR TITLE
Feature/sg 1182 support sparse class representation

### PR DESCRIPTION
**Pull request by Louis-Dupont**
### Issue
When we have a dataset with many class ids corresponding to no label at all (e.g. if only 0, 1, 255 are actual labels, and all the ids inbetween correspond to nothing, and never appear in the dataset), we currently don't have a nice way to let the user set the class names.

This is the previous "solution", to fill with dummy names:
```python
manager = ClassificationAnalysisManager(
    train_data=DataLoader(train),
    val_data=DataLoader(valid),
    class_names=["car", "person", "<unknown>", "<unknown>", ...., "<unknown>", "background"] #
)
```

### Solution
Support a mapping of id -> class_name

```python
manager = ClassificationAnalysisManager(
    train_data=DataLoader(train),
    val_data=DataLoader(valid),
    class_names={0: "car", 1: "person", 255: "background"},
)
```

It is not meant to replace `class_names_to_use`, this should include all the classes present in the dataset.
We still support list of class_name out of backward compatibility but also out of convenience - in most cases I guess people would prefer to simply provide a list.

### CODE REVIEW

File: src/data_gradients/dataset_adapters/formatters/detection.py [hunk 1]
```diff
@@ -44,7 +44,12 @@ def format(self, images: Tensor, labels: Tensor) -> Tuple[Tensor, List[Tensor]]:
         if self.class_ids_to_use is None:
             # This may trigger questions to the user, so we prefer to set it inside `former()` and not `__init__`
             # to avoid asking questions even before the analysis starts.
-            self.class_ids_to_use = [self.data_config.get_class_names().index(class_name) for class_name in self.data_config.get_class_names_to_use()]
+            classes_to_use = set(self.data_config.get_class_names_to_use())
+            print(classes_to_use)
```
Comment (shaydeci) [inline src/data_gradients/dataset_adapters/formatters/detection.py:48]: Leftover print

File: src/data_gradients/dataset_adapters/config/data_config.py [hunk 2]
```diff
@@ -117,7 +118,7 @@ def to_json(self) -> JSONDict:
             "is_batch": self.is_batch,
             "image_channels": None if self.image_channels is None else self.image_channels.channels_str,
             "n_classes": self.n_classes,
-            "class_names": self.class_names,
+            "class_id_to_name": self.class_id_to_name,
```
Comment (shaydeci) [inline src/data_gradients/dataset_adapters/config/data_config.py:121]: Why would we introduce such a big change instead of just extending the current class_names to support mappings ?

**Review by shaydeci — CHANGES_REQUESTED** [review]
The vast majority of the changes here are due to the renaming of class_names which is all over this PR.
I don't understand why we can't just extend class_names to support a mapping as well...
This really complicates the PR and introduces a breaking change.

File: src/data_gradients/dataset_adapters/formatters/detection.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/data_gradients/dataset_adapters/formatters/detection.py]: Removed [3685fab](https://github.com/Deci-AI/data-gradients/pull/205/commits/3685fab1d871573358e8fa9af819221006a37f3e)

File: src/data_gradients/dataset_adapters/config/data_config.py (hunk 2 above)
Comment (Louis-Dupont) [reply src/data_gradients/dataset_adapters/config/data_config.py:121]: The change `class_id_to_name` -> `class_names ` is mainly internal (unless someone instantiates the config directly).

The reason why I changed it is because when I first left it as `class_names`, it was quite confusing because this naming convention usually refers to a list.
But yeah, I wasnt sure whether the change of name was a good idea or not, I expected you or Eugene to raise the point.

It's easy to rollback, should we?

Comment (shaydeci) [reply src/data_gradients/dataset_adapters/config/data_config.py:121]: I think we should, yes.

Comment (Louis-Dupont) [reply src/data_gradients/dataset_adapters/config/data_config.py]: Done here [e941c0d](https://github.com/Deci-AI/data-gradients/pull/205/commits/e941c0db81f34938f83a35ff1b77770ebe75f2a7)

**Review by shaydeci — APPROVED** [review]
LGTM
